server/graphql/v2/object/Host.js

```diff
@@ -590,14 +648,48 @@ export const Host = new GraphQLObjectType({
           if (dateRange) {
             where.createdAt = dateRange;
           }
+          let collectiveIds;
           if (args.account) {
             const collectives = await fetchAccountsWithReferences(args.account, { throwIfMissing: true });
             const collectiveIds = collectives.map(collective => collective.id);
             where.CollectiveId = { [Op.in]: collectiveIds };
           }
+
+          const expenseAmountOverTime = async () => {
+            let expenseAmountOverTime;
+            if (args.timeUnit) {
+              const dateFrom = args.dateFrom ? moment(args.dateFrom).toISOString() : undefined;
+              const dateTo = args.dateTo ? moment(args.dateTo).toISOString() : undefined;
+              expenseAmountOverTime = await queries.getTransactionsTimeSeries(
+                TransactionKind.EXPENSE,
+                TransactionTypes.DEBIT,
+                host.id,
+                args.timeUnit,
+                collectiveIds,
+                dateFrom,
+                dateTo,
+              );
+              expenseAmountOverTime = expenseAmountOverTime.map(expenseAmount =>
+                convertCurrencyAmount(expenseAmount.currency, host.currency, expenseAmount.date, expenseAmount.amount),
+              );
+              expenseAmountOverTime = await Promise.all(expenseAmountOverTime);
+              expenseAmountOverTime = expenseAmountOverTime.map(({ date, amount }) => {
+                return { date, amount: Math.abs(amount), currency: host.currency };
+              });
```

Comment:
We're mapping the result 2 times, could we do that in one go? And since we do almost the exact same thing for contributions, maybe we could think about a smart helper to reduce the code duplication between them?

```es6
const result = await Promise.all(expenseAmountOverTime.map(async expenseAmount => {
  return {
    amount: Math.abs(await convertCurrencyAmount(...)),
    ...
  }
}))
```

Reply:
Corrected this one by moving the logic to a helper function. Also note that I've moved the `Math.abs` to the front-end as I felt that it's the duty of the front end to present the data in the chat as they want; we now just send the expense data points as negative values. 😉